src/main/resources/db/migration/V10__fix_datasources.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE datasources ADD COLUMN urn VARCHAR(128) NOT NULL;
+ALTER TABLE datasources ADD CONSTRAINT unique_name UNIQUE (name, urn)
```

Comment:
The `urn` itself is good enough for uniqueness. The datasource URN will have the name as part of the URN:

```
urn:{datasource_type}:{datasource_name}
```